### ISSUE TITLE
Clarifying the training setup

### ISSUE THREAD

**Issue by nikky4D** [openai/CLIP#42]
I'm confused as to your training loss and setup.

For the setup, you say:

>  We remove the non-linear projection between the representation and the contrastive embedding space, a change which was introduced by Bachman et al. (2019) and popularized by Chen et al. (2020b). We use only a linear projection to map from each encoder’s representation to the multi-modal embedding space. We did not notice a difference in training efficiency between
the two versions and speculate that non-linear projections may be co-adapted with details of current image only self supervised representation learning methods.

Do you use the non-linear projection in pretraining, then remove it after training as in Chen, 2020b, replacing it with a linear projection after training? or do you use a linear projection in pretraining, and keep the linear projection after training?
And can you explain the speculation you talk about above? I don't understand what you mean there.

For the loss: 
Can you clarify that the loss used in training is the same form of the loss function as in Zheng et. al, 2021?

**Comment by jongwook** [comment]
Sorry about the confusion; you could read "we remove" as "we do not use". The projection is linear during training and evaluation.

Looks like we're missing a preposition in that sentence. The speculation was that nonlinear projections might be effective only in self-supervised methods but not in contrastive multimodal settings like CLIP.

The loss function takes the same form as ConVIRT by Zhang et al. (2020).

**Comment by nikky4D** [comment]
Thanks for the update. One last question to clarify: Your linear function is the weight matrix, the `W_i, W_t `variable in the pseudocode?

**Comment by jongwook** [comment]
Yes! That corresponds to [here](https://github.com/openai/CLIP/blob/main/clip/model.py#L232-L233), [here](https://github.com/openai/CLIP/blob/main/clip/model.py#L82), and [here](https://github.com/openai/CLIP/blob/e5347713f46ab8121aa81e610a68ea1d263b91b7/clip/model.py#L349) in the model code.

**Comment by nikky4D** [comment]
Thank you very much. I appreciate it.
